apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.html.ts

```diff
@@ -0,0 +1,256 @@
+import { HtmlSafeString, html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { z } from 'zod';
+import { CourseRequestSchema, UserSchema } from '../../lib/db-types';
+
+export const CourseRequestRowSchema = z.object({
+  course_request: CourseRequestSchema,
+  approved_by_user: UserSchema.nullable(),
+});
+type CourseRequestRow = z.infer<typeof CourseRequestRowSchema>;
+
+export function RequestCourse({
+  rows,
+  resLocals,
+}: {
+  rows: CourseRequestRow[];
+  resLocals: Record<string, any>;
+}) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head')%>", {
+          ...resLocals,
+        })}
+        <script>
+          $(function () {
+            $('input[name=cr-role]').change(function () {
+              var role = this.value;
+              $('.question-form button').prop('disabled', role != 'instructor');
+              $('.role-comment').hide();
+              $('.role-comment-' + role).show();
+            });
+          });
+        </script>
+      </head>
+      <body>
+        ${renderEjs(__filename, "<%- include('../partials/navbar')%>", {
+          ...resLocals,
+          navPage: 'request_course',
+        })}
+        <main id="content" class="container">
+          ${CourseRequestsCard({ rows })}
+          ${CourseNewRequestCard({ csrfToken: resLocals.__csrf_token })}
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function CourseRequestsCard({ rows }: { rows: CourseRequestRow[] }): HtmlSafeString {
+  if (rows.length === 0) {
+    return html``;
+  } else {
+    return html`
+      <div class="card mb-4">
+        <div class="card-header bg-primary text-white d-flex align-items-center">
+          Course Requests
+        </div>
+        <div class="table-responsive">
+          <table class="table table-sm table-hover table-striped">
+            <thead>
+              <tr>
+                <th>Short Name</th>
+                <th>Title</th>
+                <th>Status</th>
+                <th>Details</th>
+              </tr>
+            </thead>
+            <tbody>
+              ${rows.map(({ course_request, approved_by_user }) => {
+                let details = '';
+                switch (course_request.approved_status) {
+                  case 'approved':
+                    if (approved_by_user) {
+                      details = `Approved by ${approved_by_user?.name}`;
```

Comment:
Nit, this shouldn't be necessary.

```suggestion
                      details = `Approved by ${approved_by_user.name}`;
```